lego-mindstorms-pieces.py

```diff
@@ -339,67 +365,132 @@ def order(shop=None, browser=None, lego_set=None, order_list=None, username=None
-                electric_part_list.append(part_no)
-                electric_part_counter += 1
-            else:
-                print("OOOPS! No LEGO part with that number found in set #{set}. :-(".format(
-                    set=lego_set))
-                not_in_set_counter += 1
+        new_part_no_list = []
+
+        if original_part_no in newelementid_map.keys():
+            new_part_no_list = newelementid_map[original_part_no]
+
+        part_found = False
+        partno_list = list([original_part_no] + new_part_no_list)
+
+        for idx, part_no in enumerate(partno_list):
+
+            # idx 0 has the original part_no
+            if idx > 0:
+                print("\t>> Trying to replace with #{pn} ".format(pn=part_no), end='')
+
+            element_field = wait.until(EC.element_to_be_clickable((By.ID, 'element-filter')))
+            element_field.clear()
+            element_field.send_keys(part_no)
+            element_field.send_keys(Keys.RETURN)
+            sleep(.3)  # seconds
+
+            try:
+                # tip : count results to ensure the wanted part_no return nothing or one
+                results_count = len(
+                    browser.find_elements_by_css_selector('.element-details + button'))
+
+                if results_count == 0:
+
+                    if is_electric_part(part_no):
+                        print("Not Found, but electric part:")
+                        print("\t!! The LEGO Group provides electric part out of set #{set}, "
+                              .format(set=lego_set), end='')
+                        print("see note at the end.")
+                        electric_part_list.append(part_no)
+                        electric_part_counter += 1
+                        # quit the part_no loop
+                        break
+
+                    # we're on the original part, and we've a list of new Element ID
+                    if idx == 0 and len(partno_list) > 1:
+                        print("Not Found, but has a chain of other Element ID:")
+                        # a comment about the mapping
+                        if part_no in newelementid_comment.keys():
+                            print("\tcomment: {comment}".
+                                  format(comment=newelementid_comment[original_part_no]))
+
+                    else:
+                        print("Not Found!")
+
+                elif results_count == 1:
+                    # only one result, perfect !
+                    part_found = True
+                    added_part[part_no] = original_part_no
+                    print("Found!")
+
+                    add_button = browser.find_element_by_css_selector('.element-details + button')
+
+                    if add_button.is_enabled():
+                        add_button.click()
+                        sleep(.2)  # seconds
+                        found_counter += 1
+
+                        # quit the part_no chain loop
```

Comment:
This comment is a (relatively strong) indication that the loop body is significantly too long. Try moving some parts of the loop body into helper functions to make the code easier to understand. When everything is logical, even without comments, then you're good.